x/datadeal/abci.go

```diff
@@ -1,13 +1,72 @@
 package datadeal
 
 import (
+	"strconv"
+
 	sdk "github.com/cosmos/cosmos-sdk/types"
 	"github.com/medibloc/panacea-core/v2/x/datadeal/keeper"
 	"github.com/medibloc/panacea-core/v2/x/datadeal/types"
 	oracletypes "github.com/medibloc/panacea-core/v2/x/oracle/types"
 )
 
 func EndBlocker(ctx sdk.Context, keeper keeper.Keeper) {
+	keeper.IterateClosedDataVerificationQueue(ctx, ctx.BlockHeader().Time, func(dataSale *types.DataSale) bool {
+
+		keeper.RemoveDataVerificationQueue(ctx, dataSale.DealId, dataSale.VerifiableCid, dataSale.VotingPeriod.VotingEndTime)
+		iterator := keeper.GetDataVerificationVoteIterator(ctx, dataSale.DealId, dataSale.VerifiableCid)
+
+		defer iterator.Close()
+
+		oracleKeeper := keeper.GetOracleKeeper()
+
+		tallyResult, err := oracleKeeper.Tally(
+			ctx,
+			iterator,
+			func() oracletypes.Vote {
+				return &types.DataVerificationVote{}
+			},
+			func(vote oracletypes.Vote) error {
+				return keeper.RemoveDataVerificationVote(ctx, vote.(*types.DataVerificationVote))
+			},
+		)
+
+		if err != nil {
+			panic(err)
+		}
+
+		if tallyResult.IsPassed() {
+			dataSale.Status = types.DATA_SALE_STATUS_DELIVERY_VOTING_PERIOD
+			if dataSale.VerifiableCid != string(tallyResult.ConsensusValue) {
+				panic("invalid verifiable CID consensus value")
+			}
+
+			keeper.AddDataDeliveryQueue(ctx, dataSale.VerifiableCid, dataSale.DealId, oracleKeeper.GetVotingPeriod(ctx).VotingEndTime)
```

Comment:
When the data verification vote is passed, a new vote for data delivery starts.
Accordingly, we need to update the voting period :)

Reply:
Yes, I wonder about that. How to get the updated voting period?

Reply:
You can refer this part to get `votingPeriod`. (https://github.com/medibloc/panacea-core/blob/master/x/oracle/keeper/oracle.go#L378)

Umm... As we decide to vote twice, I think the votingPeriod field should also be split.
```protobuf
message DataSale {
  string seller_address = 1;
  uint64 deal_id = 2;
  string verifiable_cid = 3;
  string delivered_cid = 4;
  DataSaleStatus status = 5;
  panacea.oracle.v2alpha2.VotingPeriod verification_voting_period = 6;
  panacea.oracle.v2alpha2.VotingPeriod delivery_voting_period = 7;
  panacea.oracle.v2alpha2.TallyResult verification_tally_result = 8;
  panacea.oracle.v2alpha2.TallyResult delivery_tally_result = 9;
}
```
What do you think?

Reply:
Thanks @gyuguen!
Yes we can do. Or can set newly updated `votingPeriod` to data sale and use it.
Both are ok to me.

Reply:
Yes, I also used a `GetVotingPeriod()` method. So, I'll change to `dataSale.DeliveryVotingPeriod.VotingEndTime`.

Reply:
I think this part can be refactored quite a lot. So I'll make a new branch and work on there.

Reply:
OK

Reply:
This comments reflected in PR #456. Please refer to the branch.